source/utilities.tex

```diff
@@ -3383,6 +3395,179 @@
 \end{codeblock}
 \end{itemdescr}
 
+\rSec3[optional.monadic]{Monadic operations}
+
+\indexlibrarymember{and_then}{optional}
+\begin{itemdecl}
+template<class F> constexpr auto and_then(F&& f) &;
+template<class F> constexpr auto and_then(F&& f) const&;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+Let \tcode{U} be \tcode{invoke_result_t<F, decltype(value())>}.
+
+\pnum
+\mandates
+\tcode{remove_cvref_t<U>} is a specialization of \tcode{optional}.
+
+\pnum
+\effects
+Equivalent to:
+\begin{codeblock}
+if (*this) {
+  return invoke(std::forward<F>(f), value());
+} else {
+  return remove_cvref_t<U>();
+}
+\end{codeblock}
+\end{itemdescr}
+
+\indexlibrarymember{and_then}{optional}
+\begin{itemdecl}
+template<class F> constexpr auto and_then(F&& f) &&;
+template<class F> constexpr auto and_then(F&& f) const&&;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+Let \tcode{U} be \tcode{invoke_result_t<F, decltype(std::move(value()))>}.
+
+\pnum
+\mandates
+\tcode{remove_cvref_t<U>} is a specialization of \tcode{optional}.
+
+\pnum
+\effects
+Equivalent to:
+\begin{codeblock}
+if (*this) {
+  return invoke(std::forward<F>(f), std::move(value()));
+} else {
+  return remove_cvref_t<U>();
+}
+\end{codeblock}
+\end{itemdescr}
+
+\indexlibrarymember{transform}{optional}
+\begin{itemdecl}
+template<class F> constexpr auto transform(F&& f) &;
+template<class F> constexpr auto transform(F&& f) const&;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+Let \tcode{U} be \tcode{remove_cv_t<invoke_result_t<F, decltype(value())>>}.
+
+\pnum
+\mandates
+\tcode{U} is a non-array object type
+other than \tcode{in_place_t} or \tcode{nullopt_t}.
+The declaration
+\begin{codeblock}
+U u(invoke(std::forward<F>(f), value()));
+\end{codeblock}
+is well-formed for some invented variable \tcode{u}.
+\begin{note}
+\tcode{is_move_constructible_v<U>} can be \tcode{false}.
```

Comment:
I guess we should harmonize this with the wording we settled on in #4732 - it's trying to call out the same thing. Here we already know that `invoke(std::forward<F>(f), value())` is a prvalue of type `cv U` by definition, so perhaps:

```suggestion
There is no requirement that \tcode{U} is movable\iref{dcl.init.general}.
```

Reply:
Fixed